### PR TITLE
[main] Second batch of migrations to enable `libguestfs`.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Second batch of packages needed to unblock `libguestfs`. Most changes are a basic clean-up and license verification. See the change log below for details regarding packages, which required more work.

###### Change Log  

- Migrated 37 packages from `SPECS-EXTENDED`.
- Fixed invalid source URLs.
- Verified licenses.
- Fixed invalid version information in `ocaml-num`.
- Created a source tarball with build configurations from previous location of the sources for `ocaml-fileutils` to make them build correctly with the proper GitHub sources (old URL was no longer valid).

###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local package builds.

### CODE REVIEW

File: SPECS/ocaml-base/ocaml-base.spec [hunk 1]
```diff
@@ -1,18 +1,16 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
 # TESTING NOTE: The ppx_jane module is needed to run the tests.  However,
 # ppx_jane transitively requires this module.  Therefore, we cannot run the
 # tests at all until we are able to add ppx_jane to Fedora, and even then we
```
Comment (oliviacrain) [inline SPECS/ocaml-base/ocaml-base.spec:3]: Nit: Fedora-specific comment here that could be removed

File: SPECS/ocaml-alcotest/ocaml-alcotest.spec [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
-Release:        2%{?dist}
-Summary:        Lightweight and colorful test framework for OCaml
-
+Release:        3%{?dist}
 License:        ISC
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 URL:            https://github.com/mirage/alcotest
 Source0:        %{URL}/archive/%{version}.tar.gz#/%{srcname}-%{version}.tar.gz
 # We neither need nor want the stdlib-shims package in Fedora.  It is a forward
```
Comment (oliviacrain) [inline SPECS/ocaml-alcotest/ocaml-alcotest.spec:17]: Nit: Fedora-specific comment

File: SPECS/ocaml-cinaps/ocaml-cinaps.spec [hunk 3]
```diff
@@ -1,22 +1,20 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
+%global srcname cinaps
+
 # TESTING NOTE: The ppx_jane module is needed to run the tests.  However,
 # ppx_jane transitively requires this module.  Therefore, we cannot run the
 # tests at all until we are able to add ppx_jane to Fedora, and even then we
```
Comment (oliviacrain) [inline SPECS/ocaml-cinaps/ocaml-cinaps.spec:5]: Nit: Fedora-specific comment

File: SPECS/ocaml-fmt/ocaml-fmt.spec [hunk 4]
```diff
@@ -1,17 +1,16 @@
...
-Release:        3%{?dist}
-Summary:        OCaml Format pretty-printer combinators
-
+Release:        4%{?dist}
 License:        ISC
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 URL:            https://erratique.ch/software/fmt
 Source0:        https://github.com/dbuenzli/fmt/archive/v%{version}/%{srcname}-%{version}.tar.gz
 # We neither need nor want the stdlib-shims package in Fedora.  It is a forward
```
Comment (oliviacrain) [inline SPECS/ocaml-fmt/ocaml-fmt.spec:16]: Nit: Fedora-specific comment

File: SPECS/ocaml-ounit/ocaml-ounit.spec [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
-Release:        5%{?dist}
-Summary:        Unit test framework for OCaml
-
+Release:        6%{?dist}
 License:        MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 URL:            https://github.com/gildor478/ounit
 Source0:        %{url}/releases/download/v%{version}/%{srcname}-v%{version}.tbz
 # We neither need nor want the stdlib-shims package in Fedora.  It is a forward
```
Comment (oliviacrain) [inline SPECS/ocaml-ounit/ocaml-ounit.spec:21]: Nit: Fedora-specific comment

File: SPECS/ocaml-topkg/ocaml-topkg.spec [hunk 6]
```diff
@@ -1,22 +1,20 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
+%global srcname topkg
+
 # BOOTSTRAP NOTE: currently we do not build the optional topkg-care part.
 # It has dependencies which do not yet exist in Fedora, and which themselves
 # depend on the main part of this package.  Once the dependencies have been
 # added to Fedora, we will build topkg-care in non-bootstrap builds.
```
Comment (oliviacrain) [inline SPECS/ocaml-topkg/ocaml-topkg.spec:6]: Nit: Fedora-specific comment

File: SPECS/ocaml-uutf/ocaml-uutf.spec [hunk 7]
```diff
@@ -1,17 +1,16 @@
...
-Release:        14%{?dist}
-Summary:        Non-blocking streaming Unicode codec for OCaml
-
+Release:        15%{?dist}
 License:        ISC
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
 URL:            https://erratique.ch/software/uutf
 Source0:        https://github.com/dbuenzli/uutf/archive/v%{version}/%{srcname}-%{version}.tar.gz
 # The uchar package is a forward compatibility package for OCaml versions prior
```
Comment (oliviacrain) [inline SPECS/ocaml-uutf/ocaml-uutf.spec:16]: Nit: Fedora-specific comment

File: cgmanifest.json [hunk 8]
```diff
@@ -14698,16 +14708,6 @@
         }
       }
     },
-    {
```
Comment (oliviacrain) [inline cgmanifest.json:14701]: Looks like there's some merge weirdness here with the `jna` changes.

**Review by oliviacrain — CHANGES_REQUESTED** [review]
Thanks for lifting these specs! Everything looks good overall, just two notes:

1) There are some Fedora-specific comments left over that could be re-evaluated for their applicability to Mariner.
2) Are all of the cgmanfiest changes intentional? There are some non-ocaml entries being touched there.

File: cgmanifest.json (hunk 8 above)
Comment (PawelWMS) [reply cgmanifest.json:14701]: This is fine, the tooling we have for updating the `cgmanifest.json` will always sort the entries as well. The `jna` change had some manual changes in it and they were out of order, so this got fixed here.

File: SPECS/ocaml-base/ocaml-base.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/ocaml-base/ocaml-base.spec:3]: I really appreciate catching lingering Fedora-specific comments. I managed to catch most of the macros, which we don't use in Mariner, but comments were a bit more tricky. I'll update them soon.

**Review by oliviacrain — APPROVED** [review]
Thanks for addressing my comments- approved!

**Pull request merged**